homeassistant/components/ituran/config_flow.py

```diff
@@ -0,0 +1,129 @@
+"""Config flow for Ituran integration."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from pyituran import Ituran
+from pyituran.exceptions import IturanApiError, IturanAuthError
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_REAUTH, ConfigFlow, ConfigFlowResult
+
+from .const import (
+    CONF_ID_OR_PASSPORT,
+    CONF_MOBILE_ID,
+    CONF_OTP,
+    CONF_PHONE_NUMBER,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ID_OR_PASSPORT): str,
+        vol.Required(CONF_PHONE_NUMBER): str,
+        vol.Optional(CONF_MOBILE_ID): str,
+    }
+)
+
+STEP_OTP_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_OTP): str,
+    }
+)
+
+
+class IturanConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ituran."""
+
+    _user_info: dict[str, Any]
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the inial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            await self.async_set_unique_id(user_input[CONF_ID_OR_PASSPORT])
+            if self.source == SOURCE_REAUTH:
+                self._abort_if_unique_id_mismatch(reason="id_mismatch")
+            else:
+                self._abort_if_unique_id_configured()
```

Comment:
```suggestion
            self._abort_if_unique_id_configured()
```
nitpicky: we don't need the `else`

Reply:
or I am wrong and it did make a difference, sorry 😂

Reply:
This doesn't work as expected. I was worried about it but thought `_abort_if_unique_id_configured()` knows how to handle reauth flows. If we're in reauth, then the unque_id already exists so if I call the `_abort_if_unique_id_configured()` when in reauth, it aborts since the same ID is already configured.